### PR TITLE
168 event schema metadata

### PR DESCRIPTION
**Pull request by druminski**

### CODE REVIEW

**Comment by pbetkier** [comment]
JsonMessageContentWrapper still converts the message into {"metadata": ..., "message": ...} structure instead of adding __metadata map to the existing structure like AvroMessageContentWrapper now does. We should keep the wrapping structure consistent across formats.

Other than that, nice code :)

**Comment by adamdubiel** [comment]
I don't think we should change existing structure, at least not without some in-depth analysis of potential side effects (like how to deploy new version without breaking compatibility).

**Comment by pbetkier** [comment]
Good point, let's have such analysis and make a migration plan. I just want to make sure the structures in these formats converge at some point in the near future. I believe that's what our users (i.e. hermes adopters) would expect, not caring about the story of hermes deployment at allegro.

**Comment by pbetkier** [comment]
:+1:

File: hermes-common/src/main/java/pl/allegro/tech/hermes/common/message/wrapper/MessageContentWrapperDispatcher.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package pl.allegro.tech.hermes.common.message.wrapper;
+
+import org.apache.avro.Schema;
+import pl.allegro.tech.hermes.api.Topic;
+import pl.allegro.tech.hermes.domain.topic.schema.SchemaRepository;
+
+import javax.inject.Inject;
+
+public class MessageContentWrapperDispatcher {
```
Comment (pawelszymczyk) [inline hermes-common/src/main/java/pl/allegro/tech/hermes/common/message/wrapper/MessageContentWrapperDispatcher.java]: this class does not dispatch any MessageContenetWrapper, why it isn't MessageContentWrapper?

File: hermes-common/src/main/java/pl/allegro/tech/hermes/common/message/wrapper/MessageMetadata.java [hunk 2]
```diff
@@ -7,11 +7,11 @@
 
 public class MessageMetadata {
 
-    private final long timestamp;
+    private final Long timestamp;
```
Comment (pawelszymczyk) [inline hermes-common/src/main/java/pl/allegro/tech/hermes/common/message/wrapper/MessageMetadata.java]: it's necessary to enter this change?

File: hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/publishing/metadata/MetadataAddingMessageConverter.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package pl.allegro.tech.hermes.frontend.publishing.metadata;
+
+import pl.allegro.tech.hermes.api.Topic;
+import pl.allegro.tech.hermes.common.message.wrapper.MessageContentWrapperDispatcher;
+import pl.allegro.tech.hermes.frontend.publishing.message.Message;
+
+import javax.inject.Inject;
+
+public class MetadataAddingMessageConverter {
```
Comment (pawelszymczyk) [inline hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/publishing/metadata/MetadataAddingMessageConverter.java]: in my opinion this class is redundant, over engineered

File: hermes-test-helper/src/main/java/pl/allegro/tech/hermes/test/helper/avro/RecordToBytesConverter.java [hunk 4]
```diff
@@ -13,9 +13,9 @@
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 
-public class RecordToBytesConverter {
+public interface RecordToBytesConverter {
```
Comment (pawelszymczyk) [inline hermes-test-helper/src/main/java/pl/allegro/tech/hermes/test/helper/avro/RecordToBytesConverter.java]: AvroRecordToBytesConverter is very similiar with this class

File: hermes-common/src/main/java/pl/allegro/tech/hermes/common/message/wrapper/MessageContentWrapperDispatcher.java (hunk 1 above)
Comment (druminski) [inline hermes-common/src/main/java/pl/allegro/tech/hermes/common/message/wrapper/MessageContentWrapperDispatcher.java]: ok, class renamed

File: hermes-common/src/main/java/pl/allegro/tech/hermes/common/message/wrapper/MessageMetadata.java (hunk 2 above)
Comment (druminski) [inline hermes-common/src/main/java/pl/allegro/tech/hermes/common/message/wrapper/MessageMetadata.java]: it's not necessary, change reverted

**Comment by pawelszymczyk** [comment]
:+1:

File: hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/publishing/metadata/MetadataAddingMessageConverter.java (hunk 3 above)
Comment (druminski) [inline hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/publishing/metadata/MetadataAddingMessageConverter.java]: We need to invoke this code, question is where? I don't want to move it to PublishingServlet so I decided to not change it.

File: hermes-test-helper/src/main/java/pl/allegro/tech/hermes/test/helper/avro/RecordToBytesConverter.java (hunk 4 above)
Comment (druminski) [inline hermes-test-helper/src/main/java/pl/allegro/tech/hermes/test/helper/avro/RecordToBytesConverter.java]: Yes, they are the same but in different scopes so we need small duplication.

**Pull request merged**